plugins/by-name/oil-git-status/default.nix

```diff
@@ -0,0 +1,56 @@
+{
+  lib,
+  config,
+  ...
+}:
+lib.nixvim.plugins.mkNeovimPlugin {
+  name = "oil-git-status";
+  packPathName = "oil-git-status.nvim";
+  package = "oil-git-status-nvim";
+
+  description = ''
+    Add Git Status to oil.nvim directory listings.
+
+    Git status is added to the listing asynchronously after creating the oil directory
+
+    listing so it won't slow oil down on big repositories. The plugin puts the status in two new sign columns
+
+    the left being the status of the index, the right being the status of the working directory
+
+    `win_options.signcolumn` needs to be configured in `plugins.oil.settings`:
+    ```nix
+    plugins.oil = {
+      enable = true;
+      settings = {
+        win_options = {
+          signcolumn = "yes:2,";
+        };
+      };
+    };
+    ```
+  '';
+
+  maintainers = [ lib.maintainers.FKouhai ];
+
+  # Ensure oil-git-status is set up after oil is loaded
+  configLocation = lib.mkOrder 1100 "extraConfigLua";
+
+  dependencies = [
+    "git"
+  ];
+
+  extraConfig = cfg: {
+    warnings = lib.nixvim.mkWarnings "plugins.oil-git-status" {
+      when = !config.plugins.oil.enable;
+
+      message = ''
+        In order to use this plugin `plugins.oil` has to be enabled with the following setting
+        -> `plugins.oil.settings.win_options.signcolumn = "yes:2,";`
```

Comment:
I'd put the settings bit in another warning:

```nix
warnings = lib.nixvim.mkWarnings "plugins.oil-git-status" [
  {
    when = !config.plugins.oil.enable;
    # ...
  }
  {
    # TODO: are there other permitted values?
    # Should we do a regex match instead of equality check?
    when = config.plugins.oil.settings.win_options.signcolumn or null != "yes:2,";
    message = ''
      This plugin requires the following `plugins.oil` setting:
          plugins.oil.settings = {
            win_options.signcolumn = "yes:2,";
          }
    '';
  }
]